core/src/main/java/com/sequenceiq/cloudbreak/service/ServiceEndpointCollector.java

```diff
@@ -253,14 +253,14 @@ private String getHdfsUIUrlWithHostParameterFromGatewayTopology(String managerIp
         Gateway gateway = gt.getGateway();
         String protocol = autoTlsEnabled ? "https" : "http";
         Integer port = autoTlsEnabled ? ExposedService.NAMENODE.getTlsPort() : ExposedService.NAMENODE.getPort();
-        String url = String.format("https://%s:%s/%s/%s%s?host=%s://%s:%s", managerIp, knoxPort, gateway.getPath(), gt.getTopologyName(),
+        String url = String.format("https://%s:%s/%s/%s%s?host=%s://%s:%s/", managerIp, knoxPort, gateway.getPath(), gt.getTopologyName(),
                 ExposedService.NAMENODE.getKnoxUrl(), protocol, nameNodePrivateIp, port);
         return url;
     }
 
     private String getHBaseUIUrlWithHostParameterFromGatewayTopology(String managerIp, GatewayTopology gt, String nameNodePrivateIp) {
         Gateway gateway = gt.getGateway();
-        String url = String.format("https://%s:%s/%s/%s%s?host=%s&port=%s", managerIp, knoxPort, gateway.getPath(), gt.getTopologyName(),
+        String url = String.format("https://%s:%s/%s/%s%s?host=%s&port=%s/", managerIp, knoxPort, gateway.getPath(), gt.getTopologyName(),
```

Comment:
This trailing slash isn't correct. Should be before the `?` 

I think the whole line should be:

`String url = String.format("https://%s:%s/%s/%s%s/?host=%s&port=%s", managerIp, knoxPort, gateway.getPath(), gt.getTopologyName(),`

Reply:
Right, silly me :) 👍